W-11862525-first-draft-for-beta-site-build-CP (#114)
mulesoft/docs-apikit: modules/ROOT/pages/apikit-error-handling-reference.adoc

> = Error Handling Reference

Reviewer: I think you need the product name for SEO.

Author: OK! Thanks!

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:page-aliases: handle-errors-4-concept.adoc

> == Error Types

Reviewer: same comment as line 1

Author: OK! Thanks!

APIkit for REST defines the following error types:

[%header]
|===
| Error Type| Description
| APIKIT:BAD_REQUEST | Bad request
| APIKIT:NOT_FOUND | Resource not found
| APIKIT:METHOD_NOT_ALLOWED | Method not allowed
| APIKIT:NOT_ACCEPTABLE | Not acceptable
| APIKIT:UNSUPPORTED_MEDIA_TYPE | Unsupported media type
| APIKIT:NOT_IMPLEMENTED | Not implemented
|===

> == Error Handling

Reviewer: SEO

Author: OK! Thanks!

> APIkit scaffolder generates error-handling code based on the most widely-used HTTP status code responses, which are mapped to the error types and messages described as follows:

Reviewer: Don't hyphenate adverbs.
https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#MuleSoftCXWritingStyleReference-hyphens

Author: OK! Thanks!

[%header]
|===
| Status Code | Error Type| Response Message
| 400 | APIKIT:BAD_REQUEST | Bad request
| 404 | APIKIT:NOT_FOUND | Resource not found
| 405 | APIKIT:METHOD_NOT_ALLOWED | Method not allowed
| 406 | APIKIT:NOT_ACCEPTABLE | Not acceptable
| 415 | APIKIT:UNSUPPORTED_MEDIA_TYPE | Unsupported media type
| 501 | APIKIT:NOT_IMPLEMENTED | Not implemented
|===

== Example Error

> The following code shows an error handler for an HTTP 400 response. Take it as reference to build your own custom status codes and messages.

Reviewer: ```suggestion
The following code shows an error handler for an HTTP 400 response, which you can use to build your own custom status codes and messages.
```

Author: I like it! Thanks!

[source,xml,linenums]
----
...
    <on-error-propagate type="APIKIT:BAD_REQUEST" doc:name="On Error Propagate">
        <ee:transform xmlns:ee="http://www.mulesoft.org/schema/mule/ee/core" xsi:schemaLocation="http://www.mulesoft.org/schema/mule/ee/core http://www.mulesoft.org/schema/mule/ee/core/current/mule-ee.xsd">
            <ee:message>
                <ee:set-payload><![CDATA[%dw 2.0
    output application/json
    ---
    {message: "Bad request"}]]></ee:set-payload>
            </ee:message>
            <ee:variables>
                <ee:set-variable variableName="httpStatus">400</ee:set-variable>
            </ee:variables>
        </ee:transform>
    </on-error-propagate>
...
----

> The main flow transmits an HTTP error response that encompasses an HTTP status code paired with an associated message expressed in clear and straightforward language.

Reviewer: Which message is clear and straightforward? I don't see the referenced message.

Author: Requesting assistance from SMEs. Thanks!

Author: The clear and straightforward message is the one included in the body of the message. In this case, it's the “message: “Bad request”" that appears in the body.

= APIkit for REST Error Handling Reference
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:page-aliases: handle-errors-4-concept.adoc

> APIkit creates error-handling code using Mule 4 main mechanism. When something goes wrong with a part of the process, you receive a typed error message. This code covers common HTTP status codes, matched to types and error messages listed in this article.

Reviewer: Is "Mule 4 main mechanism" a common concept?
Is "typed error message" a common concept?
What does "This code" refer to? Is that the code in the error message?
"this article" is self-referential.

Author: Thanks for your feedback!
The introduction was rewritten to make it simpler.
